### PR TITLE
Update spec2 to 4.3.5

### LINKED ISSUE #91: Bump specs2 to 4.3.4

### LINKED ISSUE #100: Remove DisjunctionMatchers
The trait `DisjunctionMatchers` which is used in the tests `CacheSpec ` and `DarkSkyCacheSpec ` can be removed. The trait is not used in the mentioned test. 

In addition this prevents the adding of a dependency on `specs2-scalaz`, because the trait got moved into this module stating with version 4.0.0 of spec2.

### PR DESCRIPTION
**Pull request by mzuehlke**
- Bump specs2 to 4.3.4 (closes #91)
- Remove DisjunctionMatchers (closes #100)

it seems that #95 got stale

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -30,6 +30,7 @@ object Dependencies {
     val lruMap          = "com.snowplowanalytics" %% "scala-lru-map"        % V.lruMap
     // Tests
     val specs2          = "org.specs2"            %% "specs2-core"          % V.specs2 % "test"
+    val specs2Scalaz    = "org.specs2"            %% "specs2-scalaz"        % V.specs2 % "test"
```
Comment (BenFradet) [inline project/Dependencies.scala]: why is specs2-scalaz required?

Comment (mzuehlke) [reply project/Dependencies.scala]: With spec2 4.0.0 they move `DisjunctionMatchers` to that module.
https://github.com/etorreborre/specs2/releases/tag/SPECS2-4.0.0

It seems the trait is there (`CacheSpec` and `DarkSkyCacheSpec`) but not used. So the trait and the dependency could be removed. I will push a followup commit to this PR.

Comment (BenFradet) [reply project/Dependencies.scala]: Cool, could you log the corresponding issue and format your commits like:
- "Bump specs2 to 4.3.4 (closes #91)"
- "Remove DisjunctionMatchers (closes #xx)"

Comment (mzuehlke) [reply project/Dependencies.scala]: are two commits in a single PR ok, or should I make 2 PRs out of this ?

Comment (BenFradet) [reply project/Dependencies.scala]: 2 commits in this PR are fine, I'll rebase and merge :+1:

**Comment by snowplowcla** [comment]
Confirmed! @mzuehlke  has signed the Individual Contributor License Agreement. Thanks so much

**Comment by BenFradet** [comment]
can you rebase against develop to resolve the conflicts?

**Comment by mzuehlke** [comment]
I rebased the PR against develop

**Comment by BenFradet** [comment]
As part of [hacktoberfest](https://hacktoberfest.digitalocean.com/), we have a raffle to win Snowplow t-shirts.
To enter the raffle you can fill out this form: https://docs.google.com/forms/d/e/1FAIpQLSeBbwQMWD9w9CEvjKt1KblDVpMy9mHOvQXS6YMRYH3geM8pCw/viewform

**Pull request merged**